tests/integration/build_ssh/test_build_ssh.py

```diff
@@ -0,0 +1,226 @@
+# SPDX-License-Identifier: GPL-2.0
+
+import os
+import socket
+import struct
+import threading
+import unittest
+
+from cryptography.hazmat.primitives import serialization
+from cryptography.hazmat.primitives.asymmetric.ed25519 import Ed25519PrivateKey
+
+from tests.integration.test_podman_compose import podman_compose_path
+from tests.integration.test_podman_compose import test_path
+from tests.integration.test_utils import RunSubprocessMixin
+
+
+class TestBuildSsh(unittest.TestCase, RunSubprocessMixin):
+    def test_build_ssh(self):
+        """The build context can contain the ssh authentications that the image builder should
+        use during image build. They can be either an array or a map.
+        """
+
+        compose_path = os.path.join(test_path(), "build_ssh/docker-compose.yml")
+        sock_path = os.path.join(test_path(), "build_ssh/agent_dummy.sock")
+        private_key_file = os.path.join(test_path(), "build_ssh/id_ed25519_dummy")
+
+        agent = MockSSHAgent(private_key_file)
+
+        try:
+            # Set SSH_AUTH_SOCK because `default` expects it
+            os.environ['SSH_AUTH_SOCK'] = sock_path
+
+            # Start a mock SSH agent server
+            agent.start_agent(sock_path)
+
+            self.run_subprocess_assert_returncode([
+                podman_compose_path(),
+                "-f",
+                compose_path,
+                "build",
+                "test_build_ssh_map",
+                "test_build_ssh_array",
+            ])
+
```

Comment:
I think it would make sense to produce some output that shows that information was really acquired from the agent and then assert it in the test.

Reply:
do you mean not having the mock agent at all?

Reply:
sorry, got it, I found a solution to check the key

Reply:
Sorry for not being explicit, but your solution was pretty similar to what I was thinking about.

A more simplier solution would be to do:
```
RUN echo "No id:" >> /result.log 
RUN --mount=type=ssh ssh-add -L >> /result.log

RUN echo "id1:" >> /result.log 
RUN --mount=type=ssh,id=id1 ssh-add -L >> /result.log

RUN echo "id2:" >> /result.log 
RUN --mount=type=ssh,id=id2 ssh-add -L >> /result.log
```

Then in docker compose set command to `cat /result.log` and then check the logs in the test via `podman compose logs` (there are a few tests that do this to use for reference).

Reply:
Done